tools/mason/mason.chpl

```diff
@@ -0,0 +1,152 @@
+
+use FileSystem;
+use Help;
+use Spawn;
+use TOML;
+use MasonNew;
+use MasonBuild;
+use MasonUpdate;
+
+
+proc main(args: [] string) {
+  if args.size < 2 {
+    masonHelp();
+    exit();}
+
+  select (args[1]) {
+  when 'new' do masonNew(args);
+  when 'build' do masonBuild(args);
+  when 'update' do UpdateLock();
+  when 'run' do masonRun(args);
+  when '-h' do masonHelp();
+  when '--help' do masonHelp();
+  when '-V' do printVersion();
+  when '--version' do printVersion();
+  otherwise {
+    writeln('error: no such subcommand');
+    writeln('try mason --help');
+    exit();}
+  }
+}
+
+
+proc masonNew(args) {
+  if args.size < 3 {
+    writeln('error: Invalid arguments.');
+    masonNewHelp();
+    exit();
+  }
+  else {
+    select (args[2]) {
+    when '--help' do masonNewHelp();
+    when '-h' do masonNewHelp();
+    otherwise {
+      InitProject(args);
+    }
+    }
+  }
+}
+
+
+
+proc masonBuild(args) {
+  if args.size > 2 {
+    select (args[2]) {
+    when '-h' do masonBuildHelp();
+    when '--help' do masonBuildHelp();
+    }
+  }
+  UpdateLock();
+  BuildProgram();
+}
+
+/* TODO: mtime so compilation only when nessescary
+   TODO: eliminate need for main name  */
+proc masonRun(args) {
+  var toRun = basename(getEnv('PWD'));
+  if args.size > 2 {
+    if args[2] == "build" {
+      masonBuild(args);
+    }
+  }
+  if isFile("Mason.lock") {
+    var command = "target/debug/" + toRun;
+    runCommand(command);
+  }
+  else {
+    writeln("call mason run from the top level of your projects directory");
+  }
+}
+
+// TODO
+proc masonInit(args) {}
+proc masonClean(args) {}
+proc masonDoc(args) {}
+
+proc runCommand(cmd) {
+   var splitCmd = cmd.split();
+   var process = spawn(splitCmd, stdout=PIPE);
+   process.wait();
+   
+   for line in process.stdout.lines() {
```

Comment:
Do we *always* want to print the command output? Perhaps this should be an option in the `runCommand()` args.